### ISSUE TITLE
Can't get source-maps to work correctly?

### ISSUE THREAD

**Issue by girvo** [garthenweb/bubleify#1]
Hey there, lovely work on the browserify transform! 

I'm using the following `package.json`:

``` json
{
  "name": "skate-test",
  "version": "1.0.0",
  "description": "",
  "main": "src/index.js",
  "scripts": {
    "build": "browserify -d -g uglifyify client/app.js | exorcist build/app.js.map > build/app.js",
    "watch": "watchify -d -g uglifyify -o 'exorcist build/app.js.map > build/app.js' client/app.js",
    "dev": "nodemon -e js,html --ignore client/ src/index.js",
    "start": "node src/index.js",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Josh Girvin <josh@jgirvin.com>",
  "license": "UNLICENSED",
  "dependencies": {
    "koa": "^1.2.0",
    "koa-mount": "^1.3.0",
    "koa-static": "^2.0.0",
    "skatejs": "^0.15.3",
    "skatejs-dom-diff": "^0.3.1",
    "skatejs-named-slots": "^0.1.5"
  },
  "devDependencies": {
    "browserify": "^13.0.1",
    "bubleify": "^0.2.0",
    "exorcist": "^0.4.0",
    "nodemon": "^1.9.2",
    "uglifyify": "^3.0.1",
    "watchify": "^3.7.0"
  },
  "browserify": {
    "transform": [
      "bubleify",
      "uglifyify"
    ]
  }
}
```

And source maps are generated correctly despite `uglifyify` but it doesn't appear to work with `bubleify`? Any ideas?

The whole repo is [found here](https://github.com/girvo/skatejs-buble-docker)

**Comment by garthenweb** [comment]
Hey, after some debugging I found out that this is related to an issues already reported to substack/node-browserify#1463. There is also an open pull request that might fix it: substack/module-deps#109.

Because both, issue and pull request, are round about 5 month old, I like to think about adding a `sourceMap` option to bubleify that also allows to generates a source map. I would also like to dig deeper into the pull request to validate it, probably we can get in contact with @substack to get this merged. I will try to have some time on this on the weekend!

Nevertheless, for the moment you can work around this issue by
- adding options to the bubleify transform like `"browserify": { "transform": [ ["bubleify", { "_flags": { "debug": true } } ] ] }` in your package.json or
- move the transforms into the npm script like `browserify -b -t bubleify client/app.js -o build/app.js`.

Let me know if that fixes your issue.

**Comment by girvo** [comment]
Neither of those worked unfortunately :( 

``` javascript
  "scripts": {
    "build": "browserify client/index.js -t bubleify -d -g uglifyify | exorcist web/assets/bundle.js.map > web/assets/bundle.js",
  // ...
```

or:

``` javascript
  "browserify": {
    "transform": [
      ["bubleify", { "_flags": { "debug": true } }],
  // ...
```

Now it's no real stress on my end: To be honest, the `uglifyify` source-map is way more important, and because Bublé transforms source-to-source the ES5 it generates is super readable, but I thought I'd report it anyway in case others are wondering if they're losing it! :)

**Comment by garthenweb** [comment]
Okay, maybe there is another thing going wrong in there. If there is an issue wie bubleify I would really like to fix it.
I also had a look into your repository but I was not able to get it successfully running because the package.json file was missing and the one provided in this issue does not seam to match the directory structure in your repo. Can you please provide a working example so I can check what exactly is going wrong?

**Comment by girvo** [comment]
@garthenweb If you want to run it locally (using `node` v5+), clone it down, `cd code/`, `npm install`, then `npm run dev` and `npm run watch`: http://localhost:8080 and you should be good to go 👍

The folder structure is mapped and served via `koa` 😄

**Comment by garthenweb** [comment]
Thanks! I just missed the package.json file in the subdirectory.

Further debugging shows, that uglifyify does not accept the inline source map string used generated by buble. I created a PR that should fix it: https://github.com/hughsk/uglifyify/pull/48

You may want to install uglifyify from the branch I created to validate this as long as the PR is not merged: https://github.com/garthenweb/uglifyify/tree/AllowSourcemapUTF8Charset

**Comment by garthenweb** [comment]
I just published v0.3.0 to npm that enables source maps by default. This will solve your problem without any work around.
Because the uglifyify issue is not directly related to bubleify I will close this issue.

**Comment by girvo** [comment]
You are an absolute champion. Thanks so much!
